subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformIncrementalIntegrationTest.groovy

```diff
@@ -54,10 +54,10 @@ class ArtifactTransformIncrementalIntegrationTest extends AbstractDependencyReso
                 abstract InputChanges getInputChanges()
                 
                 @InputArtifact
-                abstract File getInput()
+                abstract Provider<FileSystemLocation> getInput()
```

Comment:
I have to say this looks ugly. Why does the user have to declare this again, and not us in `TransformAction`?

Reply:
We don't declare it so we can make changes like the one in this PR. It allows more flexibility for evolving the API.